Sources/Sentry/SentryTracer.m

```diff
@@ -133,6 +138,35 @@ - (void)captureTransaction
     }];
 }
 
+- (void)addMeasurements:(SentryTransaction *)transaction
+{
+    SentryAppStartMeasurement *appStartMeasurement = nil;
+
+    // Double-Checked Locking to avoid acquiring unnecessary locks
+    if (SentrySDK.appStartMeasurement != nil) {
+        @synchronized(SentrySDK.appStartMeasurement) {
```

Comment:
probably not an issue but if after L146 another thread sets `appStartMeasurement` to null, you crash on line L147, maybe create a dedicated locker/object?

Reply:
It wouldn't crash, but not work. Good catch thanks.